src/dawn/Optimizer/StencilInstantiation.cpp

```diff
@@ -879,7 +879,14 @@ class StencilDescStatementMapper : public ASTVisitor {
   }
 
   void visit(const std::shared_ptr<BoundaryConditionDeclStmt>& stmt) override {
-    DAWN_ASSERT_MSG(0, "BoundaryConditionDeclStmt not yet implemented");
+    BoundaryConditions bc;
+    bc.functor = stmt.get()->getFunctor();
+    // Extracting all the stencil_function arguments except for the field we apply it to
+    for(int i = 1; i < (*stmt).getFields().size(); ++i) {
+      bc.arguments.push_back(stmt.get()->getFields()[i].get()->Name);
+    }
+    // store it with the fieldname to apply it to as the key
+    instantiation_->getBoundaryConditions().emplace(stmt.get()->getFields()[0].get()->Name, bc);
```

Comment:
I know this is a common used pattern we have in the code, but I would avoid it in the future, meaning we have a getter, but in reality we are inserting members,

I would rather have 
```instantiation_->insertBoundaryCondition```
like that is clear when we want to search for when the creation and insertion of BC happens